src/hooks/useGlobal.ts

```diff
@@ -0,0 +1,33 @@
+declare global {
+  interface Window {
+    AudiusStems: any
+  }
+}
+
+window.AudiusStems = window.AudiusStems || {}
+
+/**
+ * Hook to "share state" between components using the global window object.
+ * Obviously, comes with caveats with globals.
+ *
+ * @param name shared name between users of a useGlobal
+ * @param initialValue
+ * @returns getter, setter
+ *  Similar to useState, except the getter is a function and the
+ *  setter should only be invoked with a mutator function rather than a "new value"
+ */
+const useGlobal = <T>(
+  name: string,
+  initialValue: T
+): [() => T, (mutator: (cur: T) => void) => void] => {
+  window.AudiusStems[name] = initialValue
```

Comment:
Won't this run on every re-render though? I think this needs to be in a useEffect or something so we only set the default value on the first run

Reply:
Gooooooooood catch. thx